### ISSUE TITLE
[feature][processing] data types support for map layer parameter.

### ISSUE THREAD

**Issue by QGIS-feature-tracker** [qgis/QGIS-Documentation#5020]
Original commit: https://github.com/qgis/QGIS/commit/0181d10080dc06cdffd5d91d6828d5610250b093 by alexbruy

Allow to define accepted layer types for map layer parameters, for
example vector points and rasters.

**Comment by jdlom** [comment]
Is there an interest in describing all the parameters ? For the moment, if i'm not mistaken, there is only the numeric paremeters which are describing to introduce the concept of parameter.

**Comment by jdlom** [comment]
@DelazJ what do you think ?

**Comment by DelazJ** [comment]
Sorry I saw your question the other day but forgot to reply
> Is there an interest in describing all the parameters ?

Imho, no! I think this is of a too low level to be documented. https://docs.qgis.org/3.16/en/docs/user_manual/processing/scripts.html#input-and-output-types-for-processing-algorithms lists available types but we do not bound them to each other's (if I correctly understand what it is about here).
> For the moment, if i'm not mistaken, there is only the numeric paremeters which are describing to introduce the concept of parameter.

Which section are you referring to?

**Comment by ghtmtt** [comment]
@jdlom I agree with @DelazJ that there is no need to explicit describe all the parameters. In the doc we have a link to the API and there should be enough to have more detailed description.

**Comment by jdlom** [comment]
> Which section are you referring to?

 In this section where there is an exemple with a numerical parameter.
https://docs.qgis.org/testing/en/docs/training_manual/processing/modeler_hydro.html#more-complex-models

**Comment by DelazJ** [comment]
@jdlom Right, this one is in the training manual and configuring that input value is part of the exercise step, so it's expected.

But after I read that lesson and browsed the modeler features, I better understand what this report is about (*do not rely on commit message if you can't read code* 😠 ) and I think it can be documented. Maybe I misunderstand but this PR is about the values you find when you double-click the "Map layer" input in the modeler.

We have [this section](https://docs.qgis.org/testing/en/docs/user_manual/processing/modeler.html#definition-of-inputs) in the user manual that lists the input types and describe the common parameters. We can imagine to extend it to the custom properties of each input type: kind of "they all have description, advanced, mandatory properties, but this one can also have xxx, yyy...", maybe within a table or some matrix... Currently we display a long list that we'd then make more informative.

**Comment by jdlom** [comment]
> Maybe I misunderstand but this PR is about the values you find when you double-click the "Map layer" input in the modeler.

Yes. Indeed.

My initial question was about adding subsection that gives some details about the input parameters.
 So you want I give a try on that ?

**Comment by DelazJ** [comment]
> So you want I give a try on that ?

Yes please. and sorry for the confusion. Thanks.

**Comment by jdlom** [comment]
@DelazJ 
> maybe within a table or some matrix...

Do you have any ideas on how to present it. if you have some url or screenshot to share, I'll take it.

**Comment by DelazJ** [comment]
Not yet, sorry.
